include/cppkafka/header_list.h

```diff
@@ -254,7 +266,8 @@ HeaderType HeaderList<HeaderType>::at(size_t index) const {
     if (error != RD_KAFKA_RESP_ERR_NO_ERROR) {
         throw Exception(error.to_string());
     }
-    return HeaderType(name, BufferType(value, size));
+    //Use 'Buffer' to implicitly convert to 'BufferType'
+    return HeaderType(name, Buffer(value, size));
```

Comment:
Is this relying on `Buffer` being implicitly convertible to `BufferType`? Will this work for any `BufferType` that's not `std::string`? I'm not sure what this is achieving here, what's the reason for this change?

Reply:
Yes, exactly. Also there's the conversion into `vector<unsigned char>` which is what I needed in my case. If I don't use this conversion, I pretty much have to come up with some private methods templated on the `BufferType` in which case I would do a string and vector one, so might as well use the conversion operator. 

Also `string` class has a constructor which is (ptr, size), but `vector` only has (iter_start, iter_end). Alternatively I tried adding a `Buffer::Buffer(iter_start, iter_end)` constructor so that I can skip the conversion operator, but that gives all sorts of compiler errors due to a clash with its (ptr, size) constructor. `string` class uses some SFINEA tricks for that I believe so it was too much work in this case to make it compile. Again you decide what you prefer.